cmd/admin-rpc-server.go

```diff
@@ -57,11 +58,50 @@ func (s *adminCmd) Restart(args *AuthRPCArgs, reply *AuthRPCReply) error {
 
 // ListLocks - lists locks held by requests handled by this server instance.
 func (s *adminCmd) ListLocks(query *ListLocksQuery, reply *ListLocksReply) error {
+	if err := query.IsAuthenticated(); err != nil {
+		return err
+	}
 	volLocks := listLocksInfo(query.bucket, query.prefix, query.relTime)
 	*reply = ListLocksReply{volLocks: volLocks}
 	return nil
 }
 
+func (s *adminCmd) ReInitDisks(args *AuthRPCArgs, reply *AuthRPCReply) error {
+	if err := args.IsAuthenticated(); err != nil {
+		return err
+	}
+
+	// Get the current object layer instance.
+	objLayer := newObjectLayerFn()
+
+	if _, ok := objLayer.(*xlObjects); !ok {
+		return errors.New("not supported for non erasure-code backend")
```

Comment:
Keep this as a boolean in `adminCmd` so that we don't have to reflect like this.